### PR TITLE
Float type hints for min() and max() function

### PR DESCRIPTION
**Pull request by menno-ll**
Fixes https://github.com/vinkla/extended-acf/issues/120

Changes the typehints for the `min` and `max` field settings to be `float` instead of `int`.

Allows setting options like this:
```php
$field = Number::make( _x( 'Some label', 'Fields', 'some-textdomain' ), 'some-field-name' )
	->min( 0.5 ) // This
	->max( 1.5 ) // And this
	->step( 0.1 )
	->defaultValue( 1.0 )
	->required();
```

### CODE REVIEW

File: tests/Fields/NumberTest.php [hunk 1]
```diff
@@ -26,13 +26,13 @@ public function testType()
 
     public function testMax()
     {
-        $field = Number::make('Max')->max(10)->get();
-        $this->assertSame(10, $field['max']);
+        $field = Number::make('Max')->max(10.5)->get();
+        $this->assertSame((float) 10.5, $field['max']);
```
Comment (vinkla) [inline tests/Fields/NumberTest.php]:
```suggestion
$this->assertSame(10.5, $field['max']);
```

**Comment by menno-ll** [comment]
You were so fast fast with the response, i could not fix it that fast ;) Thanks for looking so quickly

**Comment by vinkla** [comment]
Thanks for the pull request Menno!

**Comment by menno-ll** [comment]
Please note that this change might be breaking for people now providing integers in there.

**Comment by vinkla** [comment]
Not sure if that is the case. Tried it locally and it seems to be working fine. How would it break?

```php
Number::make('Number')
            ->min(1)
            ->max(10)
            ->required()

```

**Comment by menno-ll** [comment]
Nope, you are right!

I was thinking if you force cast a variable to an int, and provide that, it would crash. Like so:

```php
$field = Number::make( _x( 'Some label', 'Fields', 'some-textdomain' ), 'some-field-name' )
	->min( (int) 0.5 )
	->max( (int) 1.5 )
	->step( 0.1 )
	->defaultValue( 1.0 )
	->required();
```

But apparently PHP converts it automatically, so there's no problem there!
So ignore my comment, my assumption was wrong.

**Pull request merged**